validator.go

```diff
@@ -28,13 +28,12 @@ type ClaimsValidator interface {
 	Validate() error
 }
 
-// validator is the core of the new Validation API. It is automatically used by
+// Validator is the core of the new Validation API. It is automatically used by
 // a [Parser] during parsing and can be modified with various parser options.
 //
-// Note: This struct is intentionally not exported (yet) as we want to
-// internally finalize its API. In the future, we might make it publicly
-// available.
-type validator struct {
+// The [NewValidator] function should be used to create an instance of this
+// struct.
+type Validator struct {
```

Comment:
Another alternative (not sure if it's better/worse) is moving this to a public package, something like `package jwtvalidate`, which gets it out of the core `package jwt`.

We then document the intended use of _that_ package.

Reply:
What worries me is the example below. For those who understand what's happening, it's great. But there's going to be a cohort of folks that will conflate claim validation with token verification, and we'll be making this _extremely_ easy.

If someone assumes that checking claims alone is sufficient for token security, they might overlook the importance of verifying the token's integrity.

Having said that, I suppose since `ParseUnverified` is already the main offender, we're not accomplishing much by hiding the validator.

I hope these are valid concerns, and I'm not just being pedantic.

### Example

```go
rawToken := `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c`
var claims jwt.MapClaims
token, _, err := jwt.NewParser().ParseUnverified(rawToken, &claims)
if err != nil {
	log.Fatal(err)
}
if err := jwt.NewValidator().Validate(token.Claims); err != nil {
	log.Fatal(err)
}
fmt.Println(claims["name"])                               // John Doe
fmt.Println(time.Unix(int64(claims["iat"].(float64)), 0)) // 2018-01-17 20:30:22 -0500 EST
fmt.Println(claims["sub"])                                // 1234567890
```

Reply:
See [above](https://github.com/golang-jwt/jwt/pull/349#discussion_r1386095589). Seems problematic because of cyclic dependencies.